test/test_client.rb

```diff
@@ -0,0 +1,133 @@
+    assert !resp.results[0].changed
+
+    resp = @client.query(@index, 'Row(x=2)')
+    assert_equal resp.results[0].row.columns, [3]
+  end
+
+  def test_client_query_row
+    @client.create_field(@index, "x")
+    @client.query(@index, 'Set(1, x=2)')
+    @client.query(@index, 'Set(3, x=2)')
+
+    resp = @client.query(@index, 'Row(x=2) Row(x=3)')
+    assert_equal resp.results.length, 2
+    assert_equal resp.results[0].type, :row
+    assert_equal resp.results[0].row.columns, [1, 3]
+    assert_equal resp.results[1].type, :row
+    assert_equal resp.results[1].row.columns, []
+  end
+
+  def test_client_query_row_attrs
+    @client.create_field(@index, "x")
+    @client.query(@index, 'Set(1, x=2)')
+    @client.query(@index, 'SetRowAttrs(x, 2, username="mrpi", active=true)')
+
+    resp = @client.query(@index, 'Row(x=2)')
+    assert_equal resp.results[0].type, :row
+    assert_equal resp.results[0].row.columns, [1]
+    assert_equal resp.results[0].row.attrs, {"active"=>true, "username"=>"mrpi"}
+  end
+
+  def test_client_query_exclude_row_attrs
+    @client.create_field(@index, "x")
+    @client.query(@index, 'Set(1, x=2)')
+    @client.query(@index, 'SetRowAttrs(x, 2, username="mrpi", active=true)')
+
+    resp = @client.query(@index, 'Row(x=2)', exclude_row_attrs: true)
+    assert_equal resp.results[0].type, :row
+    assert_equal resp.results[0].row.columns, [1]
+    assert_equal resp.results[0].row.attrs, {}
+  end
+
+  def test_client_query_col_attrs
+    @client.create_field(@index, "x")
+    @client.query(@index, 'Set(1, x=2)')
+    @client.query(@index, 'SetColumnAttrs(1, username="mrpi", active=true)')
+
+    resp = @client.query(@index, 'Row(x=2)', column_attrs: true)
+    assert_equal resp.results[0].type, :row
+    assert_equal resp.results[0].row.columns, [1]
+    assert_equal resp.column_attr_sets, {1=>{"active"=>true, "username"=>"mrpi"}}
+  end
+
+  def test_client_query_count
+    @client.create_field(@index, "x")
+    @client.query(@index, 'Set(1, x=2)')
+    @client.query(@index, 'Set(3, x=2)')
+
+    resp = @client.query(@index, 'Count(Row(x=2))')
+    assert_equal resp.results.length, 1
+    assert_equal resp.results[0].type, :uint64
+    assert_equal resp.results[0].n, 2
+  end
+
+  def test_client_query_topn
+    @client.create_field(@index, "x")
+    @client.query(@index, 'Set(1, x=2)')
+    @client.query(@index, 'Set(3, x=2)')
+    @client.query(@index, 'Set(4, x=2)')
+
+    resp = @client.query(@index, 'TopN(x)')
+    assert_equal resp.results.length, 1
+    assert_equal resp.results[0].type, :pairs
+    assert_equal resp.results[0].pairs.length, 1
+    assert_equal resp.results[0].pairs[0].id, 2
+    assert_equal resp.results[0].pairs[0].count, 3
+  end
+
+  def test_client_status
+    status = @client.status()
+    assert_equal status.state, 'NORMAL'
+    assert_equal status.nodes.length, 1
+    assert status.nodes[0].id != ""
+    assert_equal status.nodes[0].uri, URI("http://localhost:10101")
```

Comment:
When I run Pilosa locally at `127.0.0.1`, this test fails:
```
  1) Failure:
ClientTest#test_client_status [/Users/travis/u/go/src/github.com/pilosa/ruby-pilosa/test/test_client.rb:129]:
--- expected
+++ actual
@@ -1 +1 @@
-#<URI::HTTP http://127.0.0.1:10101>
+#<URI::HTTP http://localhost:10101>
```
It passes if I change the test to:
```
assert_equal status.nodes[0].uri, URI("http://127.0.0.1:10101")
```

Reply:
I'm running `pilosa server` with no arguments and I'm showing the test pass with `http://localhost:10101`. Are you running with `pilosa server -b 127.0.0.1:10101`?

Reply:
Yeah, I was running pilosa with a config file containing:
```
bind = "127.0.0.1:10101"
```
This is not really a problem, I just noted it as I was reviewing things.